components/eventing-controller/cmd/eventing-controller/main.go

```diff
@@ -110,4 +110,9 @@ func main() {
 	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
 		setupLogger.Fatalw("Failed to start controller manager", "error", err)
 	}
+
+	// unsubscribe on NATS before shutdown
+	if err := natsSubMgr.UnsubscribeAll(); err != nil {
```

Comment:
Ignoring is not a option unfortunately, because if the controller shuts down ungracefully, we won't unsubscribe before shutdown and hence will have the bound pair of subscription/consumer. In that case we don't want to omit that error.

Reply:
I will create a follow-up to address this issue.